jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/TargetImageParameters.java

```diff
@@ -33,28 +36,27 @@
 public class TargetImageParameters {
 
   private final AuthParameters auth;
+  private final Project project;
 
-  @Nullable private String image;
+  private Property<String> image;
   private Set<String> tags = Collections.emptySet();
   @Nullable private String credHelper;
 
   @Inject
-  public TargetImageParameters(ObjectFactory objectFactory) {
+  public TargetImageParameters(ObjectFactory objectFactory, Project project) {
     auth = objectFactory.newInstance(AuthParameters.class, "to.auth");
+    this.project = project;
+    image = project.getObjects().property(String.class);
+    image.set(project.provider(() -> System.getProperty(PropertyNames.TO_IMAGE)));
   }
 
   @Input
-  @Nullable
-  @Optional
-  public String getImage() {
-    if (System.getProperty(PropertyNames.TO_IMAGE) != null) {
-      return System.getProperty(PropertyNames.TO_IMAGE);
-    }
+  public Provider<String> getImage() {
```

Comment:
I think we can keep the current return type of `String`. Is there a specific reason for returning `Provider<String>`?